### PR TITLE
Trigger first promotion automatically if isTag does not exist

### PR DESCRIPTION
**Pull request by hongkailiu**
To replace https://github.com/openshift/ci-tools/pull/3270 which is closed by a bad push.

https://issues.redhat.com/browse/DPTP-2509

/cc @openshift/test-platform 

/hold

Need to watch the rolling

### CODE REVIEW

File: pkg/controller/promotionreconciler/reconciler.go [hunk 1]
```diff
@@ -267,3 +288,59 @@ func configIndexFn(in cioperatorapi.ReleaseBuildConfiguration) []string {
...
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+		ciOPConfig, err := promotionConfig(ciOperatorConfigGetter, &ist)
+		if err != nil {
+			return fmt.Errorf("failed to get promotionConfig for imagestreamtag %s/%s: %w", namespace, name, err)
+		}
+		if ciOPConfig == nil {
+			return fmt.Errorf("get nil from promotionConfig for imagestreamtag %s/%s", namespace, name)
```
Comment (smg247) [inline pkg/controller/promotionreconciler/reconciler.go:325]: nit:
```suggestion
			return fmt.Errorf("nil promotionConfig for imagestreamtag %s/%s", namespace, name)
```

File: pkg/controller/promotionreconciler/reconciler.go [hunk 2]
```diff
@@ -267,3 +288,59 @@ func configIndexFn(in cioperatorapi.ReleaseBuildConfiguration) []string {
...
+		currentHEAD, found, err := currentHEADForBranch(githubClient, ciOPConfig.Metadata, log)
+		if err != nil {
+			return fmt.Errorf("failed to get current git head for %s/%s/%s and imageStreamTag %s/%s: %w",
+				ciOPConfig.Metadata.Org, ciOPConfig.Metadata.Repo, ciOPConfig.Metadata.Branch, namespace, name, err)
+		}
+		if !found {
+			return fmt.Errorf("got 404 from github for %s/%s/%s and imageStreamTag %s/%s, this likely means the repo or branch got deleted or we are not allowed to access it",
+				ciOPConfig.Metadata.Org, ciOPConfig.Metadata.Repo, ciOPConfig.Metadata.Branch, namespace, name)
+		}
+		log.WithField("name", ist.Name).WithField("namespace", ist.Namespace).Info("Requesting prowjob creation for a missing the imagestreamtag")
```
Comment (smg247) [inline pkg/controller/promotionreconciler/reconciler.go:336]:
```suggestion
		log.WithField("name", ist.Name).WithField("namespace", ist.Namespace).Info("Requesting prowjob creation for a missing imagestreamtag")
```

**Comment by hongkailiu** [comment]
I will cancel the hold on Jan 17 (back from PTO) and watch the rolling.

**Pull request merged**